accord-core/src/main/java/accord/utils/SimpleBitSet.java

```diff
@@ -0,0 +1,348 @@
+        {
+            return bs.bits;
+        }
+
+        public static SimpleBitSet construct(long[] bits)
+        {
+            return new SimpleBitSet(bits);
+        }
+    }
+
+    final long[] bits;
+    int count;
+
+    public SimpleBitSet(int size)
+    {
+        bits = new long[(size + 63)/64];
+    }
+
+    public SimpleBitSet(int size, boolean set)
+    {
+        this(size);
+        if (set)
+        {
+            Arrays.fill(bits, 0, size / 64, -1L);
+            if ((size & 63) != 0)
+                bits[indexOf(size - 1)] = -1L >>> (64 - (size & 63));
+            count = size;
+        }
+    }
+
+    public SimpleBitSet(SimpleBitSet copy)
+    {
+        bits = copy.bits.clone();
+        count = copy.count;
+    }
+
+    SimpleBitSet(long[] bits)
+    {
+        this.bits = bits;
+        for (long v : bits)
+            count += Long.bitCount(v);
+    }
+
+    public boolean set(int i)
+    {
+        int index = indexOf(i);
+        long bit = bit(i);
+        if (0 != (bits[index] & bit))
+            return false;
+        bits[index] |= bit;
+        ++count;
+        return true;
+    }
+
+    public void setRange(int from, int to)
+    {
+        if (to <= from)
+        {
+            Invariants.checkArgument(to >= from, "to < from (%s < %s)", to, from);
+            return;
+        }
+
+        int fromIndex = from >>> 6;
+        int toIndex = (to + 63) >>> 6;
+        if (fromIndex + 1 == toIndex)
+        {
+            long addBits = (-1L >>> (64 - (to & 63))) & (-1L << (from & 63));
+            orBitsAtIndex(fromIndex,  addBits);
+        }
+        else if (count == 0)
+        {
+            bits[toIndex - 1] = -1L >>> (64 - (to & 63));
+            for (int i = fromIndex + 1, maxi = toIndex - 1; i < maxi ; ++i)
+                bits[i] = -1L;
+            bits[fromIndex] = -1L << (from & 63);
+            count = to - from;
+        }
+        else
+        {
+            orBitsAtIndex(fromIndex, -1L << (from & 63));
+            for (int i = fromIndex + 1, maxi = toIndex - 1; i < maxi ; ++i)
+            {
+                count += 64 - Long.bitCount(bits[i]);
+                bits[i] = -1L;
+            }
+            orBitsAtIndex(toIndex - 1, -1L >>> (64 - (to & 63)));
+        }
+    }
+
+    private void orBitsAtIndex(int index, long setBits)
+    {
+        long prevBits = bits[index];
+        bits[index] = setBits | prevBits;
+        count += Long.bitCount(setBits) - Long.bitCount(prevBits);
+    }
+
+    public boolean unset(int i)
+    {
+        int index = indexOf(i);
+        long bit = bit(i);
+        if (0 == (bits[index] & bit))
+            return false;
+        bits[index] &= ~bit;
+        --count;
+        return true;
+    }
+
+    public boolean get(int i)
+    {
+        int index = indexOf(i);
+        long bit = bit(i);
+        return 0 != (bits[index] & bit);
+    }
+
+    public int size()
+    {
+        return bits.length * 64;
```

Comment:
this was unexpected when I started testing, normally `size` is how many elements, not how large the storage is... `setBitCount` is dead code, and `size` is always used as follows

```
waitingOnCommit.prevSetBit(waitingOnCommit.size() - 1);
```

which could be cleaned up and done as `waitingOnCommit.prevSetBit()` where we know to check the last element?

Reply:
same as `waitingOn.prevSetBitNotBefore(waitingOn.size() - 1, keyDepsCount, -1);, `waitingOn.prevSetBitNotBefore(keyDepsCount, -1);`

Reply:
The number of elements and capacity are the same in this case - we just round up the number of requested elements. It didn't seem particularly valuable to store a separate size limit that we impose that is fractionally lower, implying additional bounds checks and larger memory footprint.

Reply:
guess where I am coming from is that only real use atm are for cases that want to talk backwards, so if we default to that then the caller doesn't have to do `.size() - 1`